pwndbg/commands/vmmap.py

```diff
@@ -54,3 +57,81 @@ def vmmap(pages_filter=None):
     print(M.legend())
     for page in pages:
         print(M.get(page.vaddr, text=str(page)))
+
+
+parser = argparse.ArgumentParser()
+parser.description = 'Add Print virtual memory map page.'
+parser.add_argument('start', help='Starting virtual address')
+parser.add_argument('size', help='Size of the address space, in bytes')
+parser.add_argument('flags', nargs='?', type=str, default='', help='Flags set by the ELF file, see PF_X, PF_R, PF_W')
```

Comment:
This could probably check whether the passed chars are in `'r', 'w', 'x'`
But the current approach with checking those inside and printing out error works too...

So its up to you ;P (unless @zachriggle or @anthraxx really wants it in argparse).

Reply:
I agree. Actually, I tried when I was writing these code but didn't find a good way to do that in `argparse`. Need someone who familiar with `argparse` to help me to do that.

Reply:
https://docs.python.org/3/library/argparse.html#choices

Reply:
I tried. If you use `choice` + `nargs='+'`, you have to type `vmmap_add ... r w x`, You have to split them with a space. But what I want is `vmmap_add ... rwx`.

Reply:
I had an **insanity** idea that put all possible permutation and combination of `rwx` to handler it.......But it is annoying when you type `-h` lol

Reply:
then just add a simple function checking that,
untested but basically use type=function:
```python
def valid_vmmap_flag(flags):
    valid_flags = ['r', 'w', 'x']
    for flag in flags:
        if flag not in valid_flags:
             raise argparse.ArgumentTypeError("%s is an invalid positive flag value" % flag)
    return value

parser.add_argument('flags', nargs='?', type=valid_vmmap_flag, default='', help='Flags set by the ELF file, see PF_X, PF_R, PF_W')
```

Reply:
Lets leave it as it is